pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -552,26 +558,42 @@ func (b *OFBridge) AddOFEntriesInBundle(addEntries []OFEntry, modEntries []OFEnt
 		return err
 	}
 
+	var sentMessages int
 	addMessage := func(entrySet []entryOperation) error {
 		if entrySet == nil {
 			return nil
 		}
 		for _, e := range entrySet {
-			msg, err := e.entry.GetBundleMessage(e.operation)
-			if err != nil {
-				return err
+			var messages []ofctrl.OpenFlowModMessage
+			var err error
+			switch e.entry.(type) {
+			case Group:
+				messages, err = e.entry.(Group).GetBundleMessages(e.operation, b.maxBucketsPertMessage)
+				if err != nil {
+					return err
+				}
+			default:
+				message, err := e.entry.GetBundleMessage(e.operation)
+				if err != nil {
+					return err
+				}
+				messages = []ofctrl.OpenFlowModMessage{message}
```

Comment:
This makes the interface's method and consumer look weird and kind of anti-polymorphism.

I think it should probably move `maxBucketsPertMessage` to `ofctrl_group.go` as a public variable so it can be mutated when testing. The method should be unified to `GetBundleMessages(e.operation) ([]ofctrl.OpenFlowModMessage, error)`

Reply:
Make sense, I'll update the PR as you said.